datafusion/proto/src/lib.rs

```diff
@@ -62,6 +84,18 @@ mod roundtrip_tests {
         Box::new(Field::new(name, dt, nullable))
     }
 
+    #[tokio::test]
+    async fn roundtrip_logical_plan() -> Result<(), DataFusionError> {
```

Comment:
It might be good to add a test that uses an extension codec, to prevent future regressions

Reply:
I added a test based on the TopK extension codec from Ballista